clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -151,236 +104,142 @@ else if (clientAuthConfig.equals("requested"))
     }
 
     @Override
-    public void validateReconfiguration(Map<String, ?> configs) {
+    public void validateReconfiguration(Map<String, ?> newConfigs) {
+        if (sslEngineBuilder == null) {
+            throw new IllegalStateException("SslFactory has not been configured.");
+        }
+        Map<String, Object> nextConfigs = new HashMap<>(sslEngineBuilder.configs());
+        copyMapEntries(nextConfigs, newConfigs, SslConfigs.RECONFIGURABLE_CONFIGS);
+        if (clientAuthConfigOverride != null) {
+            nextConfigs.put(BrokerSecurityConfigs.SSL_CLIENT_AUTH_CONFIG, clientAuthConfigOverride);
```

Comment:
Why does this need special handling?

Reply:
As you know, `clientAuthConfigOverride` functions as an override.  If it is set, the value of `SSL_CLIENT_AUTH_CONFIG` given by the user is irrelevant.  It's simplest just to overwrite the value given by the user.

Of course, since `SSL_CLIENT_AUTH_CONFIG` isn't currently reconfigurable, this isn't an issue in practice right now.  But, perhaps we will let this be reconfigured in the future.